src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,9 +1,80 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.exception.WrongInitializingException;
+import mate.academy.exception.WrongInstanceCreatingException;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
+
+    public Object getInstance(Class<?> interfaceInstance) {
+        Object classImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceInstance);
+        Field[] declaredFields = clazz.getDeclaredFields();
+        if (clazz.isAnnotationPresent(Component.class)) {
+            for (Field field : declaredFields) {
+                if (field.isAnnotationPresent(Inject.class)) {
+                    Object fieldInstance = getInstance(field.getType());
+                    classImplementationInstance = createNewInstance(clazz);
+                    try {
+                        field.setAccessible(true);
+                        field.set(classImplementationInstance, fieldInstance);
+                    } catch (IllegalAccessException e) {
+                        throw new RuntimeException("Can't initialize field " + field.getName());
+                    }
+                }
+            }
+        } else {
+            throw new WrongInitializingException("Class " + clazz.getName()
+                    + " without annotation");
```

Comment:
I think that you can simplify somehow this if-else construction with exception inside else-block